packages/models/src/lib/implementation/utils.ts

```diff
@@ -0,0 +1,78 @@
+/**
+ * Regular expression to validate a slug for categories, plugins and audits.
+ * - audit (e.g. 'max-lines')
+ * - or category in RunnerOutput (e.g. 'performance')
+ * Also validates ``and ` `
+ */
+export const slugRegex = /^[a-z0-9]+(?:-[a-z0-9]+)*$/;
+
+/**
+ * Regular expression to validate a reference to a plugin's:
+ * - audit (e.g. 'eslint#max-lines')
+ * - or audit in RunnerOutput (e.g. 'eslint#max-lines')
+ * Also validates ``and ` `
+ */
+export const refRegex =
+  /^([a-z0-9:]+(?:-[a-z0-9]*)*)#([a-z0-9]+(?:-[a-z0-9]+)*$)/;
+
+/**
+ * Regular expression to validate a reference to a plugin's:
+ * - audit (e.g. 'eslint#max-lines')
+ * - or audit in RunnerOutput (e.g. 'eslint#max-lines')
+ * - or group ref (e.g. plugin-slug#group:basics)
+ * Also validates ``and ` `
+ */
+export const refOrGroupRegex =
+  /^([a-z0-9:]+(?:-[a-z0-9]*)*)#(group:)*([a-z0-9]+(?:-[a-z0-9]+)*$)/;
```

Comment:
We only have a use case for the audit-or-group refs in categories. Groups and runner ouputs need only use the audit slug (plugin is implicit), as mentioned above.

```suggestion
/**
 * Regular expression to validate a reference to a plugin's:
 * - audit (e.g. 'eslint#max-lines')
 * - or group (e.g. lighthouse#group:performance)
 * Also validates ``and ` `
 */
export const refRegex =
  /^([a-z0-9:]+(?:-[a-z0-9]*)*)#(group:)*([a-z0-9]+(?:-[a-z0-9]+)*$)/;
```